### PR TITLE
Add model validation to location excel uploads

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Bulk location uploads were missing some validation, such as on the length of the name field. This requirement existed on the django model, but the model was never validated with `full_clean()`. This PR ensures that the model is validated before attempting to save it.
Prompted by [#244300](http://manage.dimagi.com/default.asp?244300).
@czue @esoergel

### CODE REVIEW

File: corehq/apps/locations/bulk_management.py [hunk 1]
```diff
@@ -715,6 +720,27 @@ def _check_location_names(self):
...
+            try:
+                location.db_object.full_clean(exclude=exclude_fields)
+            except ValidationError as e:
+                for field, issues in e.message_dict.iteritems():
+                    for issue in issues:
+                        errors.append(_(
+                            u"Error with location in sheet '{}', at row {}. {}: {}").format(
+                                location.location_type, location.index, field, issue
+                        ))
+
```
Comment (dimagimon) [inline corehq/apps/locations/bulk_management.py]: W293 blank line contains whitespace

File: corehq/apps/locations/bulk_management.py [hunk 2]
```diff
@@ -715,6 +720,27 @@ def _check_location_names(self):
                     )
         return errors
 
+    @memoized
+    def _check_model_validation(self):
+        errors = []
+        for location in self.locations:
```
Comment (czue) [inline corehq/apps/locations/bulk_management.py]: are there any significant performance issues with this?

**Review by czue — APPROVED** [review]
looks good, though I don't have as much context as @esoergel 

build is also failing
```
======================================================================
ERROR: test_bad_type_change (corehq.apps.locations.tests.test_bulk_management.TestTreeValidator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_bulk_management.py", line 238, in test_bad_type_change
    all_errors = validator.errors
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 10, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 127, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk_management.py", line 524, in errors
    errors.extend(self._check_model_validation())
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 10, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 127, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk_management.py", line 727, in _check_model_validation
    location.lookup_old_collection_data(self.old_collection)  # This method sets location.db_object
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk_management.py", line 199, in lookup_old_collection_data
    self.db_object = copy.copy(old_collection.locations_by_id[self.location_id])
AttributeError: 'NoneType' object has no attribute 'locations_by_id'
----------------------------------------------------------------------
```

File: corehq/apps/locations/models.py [hunk 3]
```diff
@@ -374,18 +374,18 @@ def accessible_location_ids(self, domain, user):
 
 class SQLLocation(SyncSQLToCouchMixin, MPTTModel):
     domain = models.CharField(max_length=255, db_index=True)
-    name = models.CharField(max_length=100, null=True)
+    name = models.CharField(max_length=100, null=True, blank=True)
```
Comment (NoahCarnahan) [inline corehq/apps/locations/models.py]: Will remove this blank as per Ethan's suggestion

File: corehq/apps/locations/bulk_management.py (hunk 2 above)
Comment (NoahCarnahan) [inline corehq/apps/locations/bulk_management.py]: There shouldn't be. These are the locations from the sheet, and they are already in memory. The import code loops through them a few other locations as well.

File: corehq/apps/locations/tests/test_bulk_management.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 from collections import namedtuple
 
 from django.test import SimpleTestCase, TestCase
+from mock import patch, MagicMock
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_bulk_management.py]: F401 'patch' imported but unused

File: corehq/apps/locations/tests/test_bulk_management.py [hunk 5]
```diff
@@ -192,11 +193,18 @@ def test_expansion_validators(self):
         self.assertEqual(set(to_validator('b')), set(['b', 'c']))
         self.assertEqual(set(to_validator('c')), set(['c']))
 
+class MockLocationStub(LocationStub):
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_bulk_management.py]: E302 expected 2 blank lines, found 1

**Review by esoergel — APPROVED** [review]

**Pull request merged**